### PR TITLE
[FLINK-2855] [gelly] Add documentation for the Gelly library methods

### PR DESCRIPTION
**Pull request by vasia**
This PR adds detailed documentation about the available Gelly library methods and improves the javadocs of the library methods constructors.

### CODE REVIEW

File: docs/libs/gelly_guide.md [hunk 1]
```diff
@@ -1450,4 +1450,140 @@ env.execute
...
+
+
+### Single Source Shortest Paths
+
+#### Overview
+An implementation of the Single-Source-Shortest-Paths algorithm for weighted graphs. Given a source vertex, the algorithm computes the shortest paths from this source to all other nodes in the graph.
+
+#### Details
+The algorithm is implemented using [vertex-centric iterations](#vertex-centric-iterations).
+In each iterations, a vertex sends to its neighbors a message containing the sum its current distance and the edge weight connecting this vertex with the neighbor. Upon receiving candidate distance messages, a vertex calculates the minimum distance and, if a shorter path has been discovered, it updates its value. If a vertex does not change its value during a superstep, then it does not produce messages for its neighbors for the next superstep. The computation terminates after the specified maximum number of supersteps or when there are no value updates.
```
Comment (fhueske) [inline docs/libs/gelly_guide.md]: In each iteration (-s)

File: docs/libs/gelly_guide.md [hunk 2]
```diff
@@ -1450,4 +1450,140 @@ env.execute
...
+
+The algorithm is implemented using [gather-sum-apply iterations](#gather-sum-apply-iterations).
+
+See the [Single Source Shortest Paths](#single-source-shortest-paths) library method for implementation details and usage information.
+
+
+### GSA Triangle Count
+
+#### Overview
+An implementation of the Triangle Count algorithm. Given an input graph, it returns the number of unique traingles in it.
```
Comment (fhueske) [inline docs/libs/gelly_guide.md]: traingles -> triangles

File: docs/libs/gelly_guide.md [hunk 3]
```diff
@@ -1450,4 +1450,140 @@ env.execute
...
+
+#### Details
+This algorithm operates in three phases. First, vertices select neighbors with IDs greater than theirs
+and send messages to them. Each received message is then propagated to neighbors with higher ID.
```
Comment (fhueske) [inline docs/libs/gelly_guide.md]: with higher IDs. (+s)

**Comment by fhueske** [comment]
Good to merge.
Found just three minor typos that can be fixed before merging.

**Comment by vasia** [comment]
Thanks a lot for the review @fhueske!
I'll address your comments and merge :)

**Pull request closed without merging**